src/app/backend/resource/ingress/list.go

```diff
@@ -74,11 +75,33 @@ func getEndpoints(ingress *v1beta1.Ingress) []common.Endpoint {
 	return endpoints
 }
 
+func contains(s []string, str string) bool {
+	for _, v := range s {
+		if v == str {
+			return true
+		}
+	}
+	return false
+}
+
+func getHosts(ingress *v1beta1.Ingress) []string {
+	hosts := make([]string, 0)
+	if len(ingress.Spec.Rules) > 0 {
```

Comment:
There is no need to check for length before loop. `for` does that implicitly by not iterating at all if the input array is empty.